specs/PermissionManagement.md

```diff
@@ -0,0 +1,554 @@
+Permission Management APIs
+===
+
+# Background
+
+The WebView2 team is adding support for more permission management scenarios. With the
+extended `PermissionRequested` event args and new permission APIs you will be able to:
+- Set permission for a site at any time
+- List all nondefault permissions
+- Control more permission kinds
+
+The new `ShouldPersist` property on the `PermissionRequestedEventArgs` lets you
+turn off the caching of permission state so that you can intercept all permission
+requests. By default, state set from the `PermissionRequested` event handler and default
+permission UI is cached across sessions and you may stop receiving PermissionRequested
+events for some permission kinds.
+
+The new APIs, `SetPermission` and `GetNonDefaultPermissionCollection`, provide
+all the information necessary to build a permission management page where a
+user can view and modify existing site permissions. The new permission kinds we will
+support are: local font list, automatic downloads, media autoplay, and file editing.
+
+See already supported [permission kinds](https://learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2?view=webview2-1.0.1418.22#corewebview2_permission_kind)
+and existing [event args](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2permissionrequestedeventargs?view=webview2-dotnet-1.0.1418.22).
+
+
+# Examples
+
+## Extended PermissionRequestedEventArgs: ShouldPersist
+```c#
+// In this example, the app listens to all requests and caches permission on its own
+// to decide whether to show custom UI to the user.
+void WebView_PermissionRequested(object sender, CoreWebView2PermissionRequestedEventArgs args)
+{
+    CoreWebView2Deferral deferral = args.GetDeferral();
```

Comment:
Pre-existing issue: The documentation for [PermissionRequested](https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2.permissionrequested?view=webview2-dotnet-1.0.1462.37) says

> If a deferral is not taken on the event args, the subsequent scripts are blocked until the event handler returns. If a deferral is taken, the scripts are blocked until the deferral is completed.

Are scripts really blocked? I would assume that scripts continue to run, but the promise remains unsettled until the deferral is completed.